py4DSTEM/process/diskdetection/diskdetection.py

```diff
@@ -671,60 +660,50 @@ def universal_threshold(pointlistarray, minIntensity, metric, minPeakSpacing, ma
-
     return pointlistarray
 
 
-def get_pointlistarray_hist(pointlistarray, return_hist = False, bins = 200):
-    
+def get_pointlistarray_hist(pointlistarray, return_hist = True, plot_hist = True, bins = 200):
     """
     Concatecates the Bragg peak intensities from a PointListArray of Bragg peak positions into one array and returns the counts and bins. 
     This output can be used for understanding the distribution of intensities in your dataset for universal thresholding.
     
     Accepts:
         pointlistarray      (PointListArray)
         return_hist         (bool) if True, returns histogram of intensities in the entire pointlist
+        plot_hist           (bool) if True, returns plot of histogram
         bins                (int) number of bins that the intensity values will be sorted into for histogram 
     
     Returns:
         peak_intensities    (ndarray) all detected peaks
-        
-        If return_hist==False (default), the figure is plotted and nothing is returned.
-        If return_hist==True, the figure and its one axis are returned, and can be
-        further edited.
+        If return_hist==True (default), the counts and bin edges are returned, and can be manually plotted.
+        If return_hist==False, the figure is plotted and only peak intensities are returned.
     """
-    
     assert np.all([name in pointlistarray.dtype.names for name in ['qx','qy','intensity']]), "pointlistarray coords must include coordinates: 'qx', 'qy', 'intensity'."
     assert 'qx' in pointlistarray.dtype.names, "pointlistarray coords must include 'qx' and 'qy'"
     assert 'qy' in pointlistarray.dtype.names, "pointlistarray coords must include 'qx' and 'qy'"
     assert 'intensity' in pointlistarray.dtype.names, "pointlistarray coords must include 'intensity'"
     
     first_pass = True
     for (Rx, Ry) in tqdmnd(pointlistarray.shape[0],pointlistarray.shape[1]):
-    
         pointlist = pointlistarray.get_pointlist(Rx,Ry)
-
-        
         for i in range(pointlist.length):
-                
             if first_pass:
                 peak_intensities = np.array(pointlist.data[i][2])
                 peak_intensities = np.reshape(peak_intensities, 1)
                 first_pass = False
-                
             else:
                 temp_array = np.array(pointlist.data[i][2])
                 temp_array = np.reshape(temp_array, 1)
                 peak_intensities = np.append(peak_intensities, temp_array)
- 
-    counts, bin_values = show_hist(peak_intensities, bins, return_hist = True)
-    
-    if return_hist:
-    
-        return peak_intensities, counts, bin_values
-    
-    else:
-        plt.show()
-        
+    if not return_hist:
         return peak_intensities
+    counts, bin_edges = np.histogram(peak_intensities, bins = bins, range = (np.min(peak_intensities), np.max(peak_intensities)) )
+    if not plot_hist:
+        if return_hist == True:
```

Comment:
I think this line is redundant - if we already passed the previous `if not return_hist` line, we shouldn't need to check this again